### ISSUE TITLE
Unknown error occured when setting up a target rate

### ISSUE THREAD

**Issue by samtygier** [BottlecapDave/HomeAssistant-OctopusEnergy#722]
### Describe the bug

I am trying to create a rate entry. After filling the details and pressing submit I get the message "Unknown error occurred" at the top of the form.

The interesting bit of the log is probably:
```
  File "/config/custom_components/octopus_energy/config_flow.py", line 181, in async_step_target_rate
    errors = validate_target_rate_config(user_input, account_info.account, now) if user_input is not None else {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/config/target_rates.py", line 163, in validate_target_rate_config
    if is_in_agile_darkzone(start_time, end_time):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/config/target_rates.py", line 111, in is_in_agile_darkzone
    if end_time <= start_time:
       ^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<=' not supported between instances of 'NoneType' and 'NoneType'
```

### Reproduction steps

From the settings I click "Add entity" and choose target rate. In the form I have put:
Name: lowest_30m
Hours: 0.5
Type: Continuous
MPAN (left as is)
Min time: 23:00:00
Max time 08:00:00
Offset: 00:00:00

I have played with other values and get the same error. Unless I use the wrong formatting for times, and then I get a specific message about that.

### Expected behaviour

Creates target entity

### Tariff Code

E-1R-AGILE-23-12-06-G

### Integration Version

v10.0.3

### Home Assistant Version

2024.1.5

### Fresh Install?

Fresh install

### Home Assistant Logs

```
Logger: aiohttp.server
Source: /usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py:421
First occurred: 11:47:23 PM (1 occurrences)
Last logged: 11:47:23 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 85, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 233, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 31, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 149, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 63, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 177, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 320, in async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 416, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/config_flow.py", line 181, in async_step_target_rate
    errors = validate_target_rate_config(user_input, account_info.account, now) if user_input is not None else {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/config/target_rates.py", line 163, in validate_target_rate_config
    if is_in_agile_darkzone(start_time, end_time):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/config/target_rates.py", line 111, in is_in_agile_darkzone
    if end_time <= start_time:
       ^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<=' not supported between instances of 'NoneType' and 'NoneType'
```

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by BottlecapDave** [comment]
Hello. Sorry you're receiving this issue. I believe it's because you've set the min/max times to `HH:MM:SS` when it should be `HH:MM` with an agile tariff. This will be fixed in the future, but if you change the format you shouldn't have any further issues relating to this.

**Comment by samtygier** [comment]
Thanks. I was thrown by getting an error message the first time "Offset must be in the form of HH:MM:SS with an optional negative symbol", so changed all the times to HH:MM:SS.

With min and max in HH:MM and offset in HH:MM:SS it works.

**Comment by BottlecapDave** [comment]
This should be fixed in v10.0.4
